### ISSUE TITLE
Extend "Override For..." functionality to Input Map

### ISSUE THREAD

**Issue by Streq** [godotengine/godot-proposals#5589]
### Describe the project you are working on

A game with different export configurations for the same platform, that each requires different input mappings that cannot coexist

### Describe the problem or limitation you are having in your project

I cannot configure an input mapping just for a given export feature, every export will have the same input mappings

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow an "Override For..." setting in the input mappings tab of the project settings. Which will allow different export features to have different input mappings.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Besides every input action there will be an "Override for..." which will let you change the given mappings for that specific Feature, just like it works for the General tab, this will spawn a new action called ActionName.Feature, for example "ui_cancel.ArcadeMode", with its own mappings.
This will cause exports with the ArcadeMode feature to ignore the general mappings for ui_cancel and use the ArcadeMode's instead.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

You can have a separate script that runs on the condition that the export has the given label and configures the InputMap by code, but it's rather cumbersome to iterate.

### Is there a reason why this should be core and not an add-on in the asset library?

This is a core configuration.

**Comment by dalexeev** [comment]
Is this already done in 4.0?

![](https://user-images.githubusercontent.com/47700418/195917695-163426a8-ddf6-4daf-b154-965026e9987f.png)

**Comment by Calinou** [comment]
> Is this already done in 4.0?

Yes, but I think it uses a different implementation and only works with operating system feature tags (it may even only work with `.macos`). It's worth trying with other feature tags such as `.release` just in case :slightly_smiling_face:
